services/galley/src/Galley/Data/CustomInstance.hs

```diff
@@ -0,0 +1,23 @@
+{-# LANGUAGE RecordWildCards #-}
+module Galley.Data.CustomInstance
+    ( getCustomInstance
+    , setCustomInstance
+    ) where
+
+import Imports
+import Cassandra
+import Galley.Data.Instances ()
+import Galley.Types
+
+import qualified Galley.Data.Queries  as Cql
+
+getCustomInstance :: MonadClient m => EmailDomain -> m (Maybe CustomInstance)
+getCustomInstance domain = fmap toCustomInstance <$> do
```

Comment:
I guess you could avoid the `toCustomInstance` declaration altogether:

```suggestion
getCustomInstance domain = fmap CustomInstance <$> do
```

(no strong opinion.)

Reply:
It's not that easy, since we get a tuple and need to turn it into a record. This could be done generically with a fancy typeclass, but I think there are better solutions for the underlying problem. Happy to discuss this.